classes/task.class.php

```diff
@@ -102,16 +102,20 @@ public function send_queued_submissions() {
 
         // Create each submission in Turnitin and upload submission.
         foreach ($submissions as $submission) {
-            // Skip if the course doesn't exist or the course is pending deletion.
-            if (!$DB->get_record('course_modules', array('id' => $submission->cm, 'deletioninprogress' => 0))) {
-                continue;
-            }
-
             // Reset headers.
             $this->tsrequest->set_headers();
 
             $tssubmission = new plagiarism_turnitinsim_submission($this->tsrequest, $submission->id);
 
+            // Skip if the course module doesn't exist or the course is pending deletion.
+            if (!$DB->get_record('course_modules', array('id' => $submission->cm, 'deletioninprogress' => 0))) {
```

Comment:
Once the course is deleted then ideally it should not be picked by this corn task?.
Don't we change flags once the course is deleted means 0 to 1 or 2, or can't we change the status of a submission to DELETED?

Reply:
The problem there is that deleting assignments and courses is a Moodle action, and I don't think there is any hook for the plugin code to detect this when the action is performed in Moodle. I've had a look. So the only way to detect if the submission belongs to a deleted assignment/course is when it is processed by the cron.

We could change the status to DELETED rather than ERROR I guess, but would have to make sure DELETED status is handled the same as ERROR elsewhere in the code I figured an ERROR status with a can't be processed due to deletion would be sufficient.